### PR TITLE
add contracts to repay from collateral

### PR DESCRIPTION
**Pull request by alexandr-martirosyan**

### CODE REVIEW

**Review by ytumanov — APPROVED** [review]

**Review by lastperson — CHANGES_REQUESTED** [review]

File: cli.js [hunk 1]
```diff
@@ -108,7 +152,8 @@ function printActions() {
   console.info('repayFor celo|cusd|ceur|creal for address amount stable|variable [privateKey]');
   console.info('migrate-step-2 address [privateKey]');
   console.info('liquidation-bot address [privateKey]');
-  console.info('liquidity-swap [privateKey] address celo|cusd|ceur to celo|cusd|ceur amount');
+  console.info('liquidity-swap address celo|cusd|ceur to celo|cusd|ceur amount [privateKey]');
+  console.info('repay-from-collateral address collateralAsset(celo|cusd|ceur) debtAsset(celo|cusd|ceur) debtRateMode(1|2) debtRepayAmount useFlashloan(true|false) [privateKey]');
```
[the remark this answers is not in the document]
Comment (lastperson) [reply cli.js:156]: also need to add creal.

File: cli.js [hunk 2]
```diff
@@ -789,23 +832,22 @@ async function execute(network, action, ...params) {
...
     }
 
     console.log(`Checking mToken ${mToken.options.address} for approval`);
     if (BN(await mToken.methods.allowance(user, repayAdapter.options.address).call()).lt(BN(maxCollateralAmount))) {
-      console.log(
-        'Approve UniswapAdapter',
-        (await mToken.methods.approve(repayAdapter.options.address, maxCollateralAmount).send({ from: user, gas: 2000000 }))
-          .transactionHash
-      );
+      console.log('Approve UniswapAdapter', (await mToken.methods .approve(repayAdapter.options.address, maxCollateralAmount) .send({ from: user, gas: 2000000 })).transactionHash);
```
Comment (lastperson) [inline cli.js:848]: formatting broke.

File: cli.js [hunk 3]
```diff
@@ -137,6 +137,38 @@ function buildSwapAndRepayParams(
...
+  if (rateMode !== 'stable' && rateMode !== 'variable') {
+    console.error('rateMode can be only "stable|variable"');
+    return false;
+  }
+  return true;
+}
+
+function isNumeric(num) {
+  if (isNaN(num)) {
+    console.error('invalid number');
```
Comment (lastperson) [inline cli.js:158]: Would be helpful to print which number is wrong (print num as well).

File: cli.js [hunk 4]
```diff
@@ -137,6 +137,38 @@ function buildSwapAndRepayParams(
   );  
 }
 
+function isValidAsset(asset) {
```
Comment (lastperson) [inline cli.js:140]: lets move this one inside execute(), and use `if (!tokens[asset]) ...`, otherwise we'll need to maintain multiple places with assets.

**Pull request merged**